controller/configentry_controller.go

```diff
@@ -105,15 +110,29 @@ func (r *ConfigEntryController) ReconcileEntry(
 		// The object is being deleted
 		if containsString(configEntry.GetObjectMeta().Finalizers, FinalizerName) {
 			logger.Info("deletion event")
-			// Our finalizer is present, so we need to delete the config entry
-			// from consul.
-			_, err := r.ConsulClient.ConfigEntries().Delete(configEntry.ConsulKind(), configEntry.Name(), &capi.WriteOptions{
+			// Check to see if consul has config entry with the same name
+			entry, _, err := r.ConsulClient.ConfigEntries().Get(configEntry.ConsulKind(), configEntry.Name(), &capi.QueryOptions{
 				Namespace: r.consulNamespace(req.Namespace, configEntry.ConsulNamespaced()),
 			})
+
 			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("deleting config entry from consul: %w", err)
+				return ctrl.Result{}, fmt.Errorf("getting config entry from consul: %w", err)
+			}
```

Comment:
So this shouldn't actually error out since if it is deleted then it's actually good.

Reply:
Nice catch. Updated this check to ignore notFound errors (I'm time traveling. Ignore timestamps.)

Reply:
Thanks for helping with finding the bug BTW. I really appreciate it.